tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -208,6 +210,178 @@ def setUp(self):
+            Condition.GREATER
+            if alert_rule.threshold_type == AlertRuleThresholdType.ABOVE.value
+            else Condition.LESS
+        )
+        condition_result = PRIORITY_MAP.get(alert_rule_trigger.label, DetectorPriorityLevel.HIGH)
+        detector_trigger = self.create_data_condition(
+            comparison=alert_rule_trigger.alert_threshold,
+            condition_result=condition_result,
+            type=threshold_type,
+            condition_group=detector_data_condition_group,
+        )
+
+        data_condition_group = self.create_data_condition_group(
+            organization=alert_rule.organization
+        )
+        self.create_workflow_data_condition_group(
+            condition_group=data_condition_group, workflow=workflow
+        )
+        action_filter = self.create_data_condition(
+            comparison=PRIORITY_MAP.get(alert_rule_trigger.label, DetectorPriorityLevel.HIGH),
+            condition_result=True,
+            type=Condition.ISSUE_PRIORITY_EQUALS,
+            condition_group=data_condition_group,
+        )
+        return detector_trigger, action_filter
+
+    def mock_migrate_resolve(
+        self, alert_rule, resolve_threshold, detector_dcg
+    ) -> tuple[DataCondition, DataCondition]:
+        """
+        Return the data conditions that correspond to alert rule resolution.
+        """
+        threshold_type = (
+            Condition.LESS_OR_EQUAL
+            if alert_rule.threshold_type == AlertRuleThresholdType.ABOVE.value
+            else Condition.GREATER_OR_EQUAL
+        )
+        detector_trigger = DataCondition.objects.create(
+            comparison=resolve_threshold,
+            condition_result=DetectorPriorityLevel.OK,
+            type=threshold_type,
+            condition_group=detector_dcg,
+        )
+        data_condition_group = self.create_data_condition_group(
+            organization=alert_rule.organization
+        )
+        workflow = AlertRuleWorkflow.objects.get(alert_rule=alert_rule).workflow
+        self.create_workflow_data_condition_group(
+            condition_group=data_condition_group, workflow=workflow
+        )
+        action_filter = DataCondition.objects.create(
+            comparison=DetectorPriorityLevel.OK,
+            condition_result=True,
+            type=Condition.ISSUE_PRIORITY_EQUALS,
+            condition_group=data_condition_group,
+        )
+        return detector_trigger, action_filter
+
+    def mock_migrate_integration_action(
+        self, alert_rule_trigger_action, data_condition_group
+    ) -> tuple[Action, DataConditionGroupAction, ActionAlertRuleTriggerAction]:
+        data = {
+            "type": alert_rule_trigger_action.type,
+            "sentry_app_id": alert_rule_trigger_action.sentry_app_id,
+            "sentry_app_config": alert_rule_trigger_action.sentry_app_config,
+        }
+
+        action = self.create_action(
+            required=False,
+            type=Action.Type.OPSGENIE,
```

Comment:
We wouldn't have an action that's Opsgenie type with sentry app ids and config, maybe we could have this function take in the type?